charts/cloudzero-agent/docs/deploy-validation.md

```diff
@@ -1,49 +1,62 @@
+1. `Pod initialization` - basic configuration elements are validated, such as the API key and egress reachability.
+2. `Post pod start` - the prometheus container runs the `post-start` checks, then posts a `cluster up` status to the Cloudzero API. Checks include validating the API key, capturing the Kubernetes version, inspecting the scrape configuration, checking the kube-state-metrics service, and the prometheus-node-exporter server reachability. The results are logged to the `/prometheus/cloudzero-validator.log` file in the container.
+3. `Pre pod stop` - the prometheus container runs the `pre-stop` checks (usually none), then posts a `cluster down` status to the Cloudzero API.
 
----
+Based on the above statements, it is also possible to diagnose from the perspective of the prometheus container. To inspect the logs, use the following command, replacing the pod name with that of your current deployment:
 
-# Troubleshooting
+```sh
+kubectl -n $NS exec -ti -c cloudzero-agent-server cloudzero-agent-server-766b4865dc-nrwc5 -- sh -c 'cat cloudzero-agent-validator.log'
+```
 
-The Cloudlock Agent must be able to contact both the `kubernetes metrics server` and the `prometheus node exporter`. The following diagram illustrates what you might see if communication is not possible between the `cloudzero-agent` and one of these services.
+> Remember to use the correct namespace and pod identity.
 
-![](./assets/env-validate-2-missing-node-exporter.png)
+### 4. Interpret the Results
 
-When inspecting the logs, pay attention to the readiness state of the target (in this case, the node-exporter). Additionally, note the resulting `failure` status for the check.
+In the screenshot above, notice the `checks` section. This section allows you to view the results of the configured checks. For any checks that are not passing, an error message will be captured to help diagnose the problem.
 
-In this case, it is important to identify why the `cloudzero-agent` is unable to communicate with the failing container check.
+---
 
-1. Is the missing service deployed? Use `kubectl get pods` to check if the service is deployed. If not, deploy the necessary chart. The `cloudzero-agent` helm chart can deploy the `kubernetes metrics server` and the `prometheus node exporter` by setting the `enable` flags to `true` as shown in the first image above.
+## Troubleshooting
 
-2. Is the `cloudzero-agent` deployed to the same network namespace? Linux cgroup namespaces isolate process(es) network, memory, and CPU. If the containers (pods) are in different namespaces, there is no direct way to communicate between them. Make sure to use the correct `--namespace` flag when deploying the chart.
+The Cloudlock Agent has the following requirements:
 
-If all else fails, reach out to support@cloudzero.com and provide the output, along with the output of `kubectl -n <namespace> describe all` for the deployment.
```

Comment:
`If all else fails, reach out to support@cloudzero.com`  - do we want to keep this bit? though, I assume people know how to reach out to us anyway